timesketch/tsctl.py

```diff
@@ -2354,3 +2355,158 @@ def _check_fk_orphans(
             )  # Minimal output if no orphans and not verbose
     else:
         print("\nOrphaned data check complete. Issues found as listed above.")
+
+
+@cli.command(name="export-db")
+@click.argument(
+    "filepath", type=click.Path(dir_okay=False, writable=True), required=True
+)
+def export_db(filepath):
+    """Export the database to a zip file."""
+    click.echo(f"Exporting database to {filepath}...")
+    engine = db_session.get_bind()
+    with engine.connect() as connection:
+        with zipfile.ZipFile(filepath, "w", zipfile.ZIP_DEFLATED) as zipf:
+            for table in BaseModel.metadata.sorted_tables:
+                table_name = table.name
+                try:
+                    result = connection.execute(table.select())
+                    df = pd.DataFrame(result.fetchall(), columns=result.keys())
+                    row_count = df.shape[0]
+                    click.echo(f"  Exporting table: {table_name} ({row_count} rows)")
+                    json_data = df.to_json(orient="records", date_format="iso")
+                    zipf.writestr(f"{table_name}.json", json_data)
+                except Exception as e:
+                    click.echo(f"Error exporting table {table_name}: {e!s}", err=True)
+                    click.echo("Database export failed.", err=True)
+                    raise click.Abort()
+    click.echo("Database export complete.")
+
+
+@cli.command(name="import-db")
+@click.argument("filepath", type=click.Path(exists=True, dir_okay=False))
+@click.option("--yes", is_flag=True, help="Skip confirmation.")
+def import_db(filepath, yes):
```

Comment:
How does this import handle the links to the opensearch data store? Since each timeline is linked to a specific searchindex name, this would be lost during the import and we would end up with dangling sesrchindices on the opensearch cluster?

Reply:
yes, it could kill those relationships, I will add a warning to the readme so people are aware (it would already happen today if one qoule take a plain dump of DB)

Reply:
WDYT with the new addition?